ONSdigital/ras-frontstage-api
# Us062 access eq

**Scope**
To generate a valid EQ url with encrypted token from a case id. There is also some quick refactoring in the tests as the copy paste monster had been in there

**How to test**
Prerequisite
Collection instrument table must have is classifiers set with form type and EQ_ID (e.g {"form_type":"0001","eq_id":"2"} which is QBS) Its type must also be EQ. The collection exercise for this instrument must also have its events loaded, this can be done using https://github.com/ONSdigital/rm-tools/tree/master/collex-loader. 
JSON_SECRET_KEYS and EQ_URL must also be set. (eg. export EQ_URL='' in a terminal) The values which will work in EQ Preprod can be found on SIT (eg. cf env ras-frontstage-api-sit) 

Once all this has been done you can hit the endpoint to generate the eq-url 
'/surveys/generate-eq-url?case_id=<case_id>&party_id=<party_id>' with the correct case_id/party_id for your collection instrument. This will generate the url which you can then put into another browser to make sure it goes to EQ

SIT has been used to test this and has the relevant branches for collection instrument, frontstage api and frontstage (No PR yet) and does work for Blocks (temporally changed to point at QBS)with their events dates loaded
